js/src/rive.ts

```diff
@@ -0,0 +1,965 @@
+import Runtime from '../../wasm/publish/rive.js';
+
+// #region LoopEvent
+
+// Loop types. The index of the type is the value that comes from Wasm
+export const loopTypes: string[] = ['oneShot', 'loop', 'pingPong'];
+
+/// Loop events are returned through onloop callbacks
+interface LoopEvent {
+  animation: string;
+  name: string;
+  type: number;
+}
+
+// Creates a new LoopEvent
+export let createLoopEvent = (animation: string, loopValue: number): LoopEvent => {
+  if (loopValue < 0 || loopValue >= loopTypes.length) {
+    throw 'Invalid loop value';
+  }
+  return {
+    animation: animation,
+    type: loopValue,
+    name: loopTypes[loopValue],
+  }
+}
+
+// #endregion
+
+// Maps the playback state to the Wasm enum values
+export const playbackStates = { 'play': 0, 'pause': 1, 'stop': 2 };
+
+// #region layout
+
+// Fit options for the canvas
+export enum Fit {
+  Cover     = 'cover',
+  Contain   = 'contain',
+  Fill      = 'fill',
+  FitWidth  = 'fitWidth',
+  FitHeight = 'fitHeight',
+  None      = 'none',
+  ScaleDown = 'scaleDown'
+}
+
+// Alignment options for the canvas
+export enum Alignment {
+  Center       = 'center',
+  TopLeft      = 'topLeft',
+  TopCenter    = 'topCenter',
+  TopRight     = 'topRight',
+  CenterLeft   = 'centerLeft',
+  CenterRight  = 'centerRight',
+  BottomLeft   = 'bottomLeft',
+  BottomCenter = 'bottomCenter',
+  BottomRight  = 'bottomRight'
+}
+
+// Interface for the Layout static method contructor
+export interface LayoutParameters {
+  fit: Fit,
+  alignment: Alignment,
+  minX: number,
+  minY: number,
+  maxX: number,
+  maxY: number
+}
+
+// Alignment options for Rive animations in a HTML canvas
+export class Layout {
+  public fit: Fit;
+  public alignment: Alignment;
+  public minX: number;
+  public minY: number;
+  public maxX: number;
+  public maxY: number;
+
+  constructor(
+    fit: Fit = Fit.None,
```

Comment:
When parameters of the constructor become properties of the class you can save a good amount of code like that :
```typescript
class Layout {
  constructor(
    public fit: Fit = 'none',
  ) {}
}
```
Typescript will automatically manage the boilerplate of defining the property and initializing it 🤩